### PR TITLE
Change `cmake.verbose = true` to `build.verbose = true`

### PR DESCRIPTION
**Pull request by jakirkham**
As `cmake.verbose = true` [is deprecated]( https://github.com/scikit-build/scikit-build-core/blob/9d3737fdb325580b7bb4fb544f7b48caf03fce13/docs/configuration.md?plain=1#L83-L89 ), drop it from the config

Replace it with the now preferred `build.verbose = true`

Also switch from `cmake.minimum-version` to `cmake.version`, which fixes a CI error

Require scikit-build-core 0.10.0+

### CODE REVIEW

**Review by jameslamb — APPROVED** [review]
Thanks!

Hope it's ok, I pushed a commit adding the other changes for https://github.com/rapidsai/build-planning/issues/58 here as they seemed relevant (setting a floor of `scikit-build-core >= 0.10.0`, reading the CMake minimum from `CMakeLists.txt`). Feel free to revert if you disagree, I won't merge this until you've had a chance to respond to that.

**Comment by jakirkham** [comment]
Restarting CI

File: pyproject.toml [hunk 1]
```diff
@@ -5,16 +5,17 @@ build-backend = "scikit_build_core.build"
 dependencies-file = "dependencies.yaml"
 
 [tool.scikit-build]
-cmake.minimum-version = "3.26.4"
-cmake.verbose = true
+build.verbose = true
+cmake.version = "CMakeLists.txt"
+minum-version = "0.10"
```
Comment (jakirkham) [inline pyproject.toml:10]:
```suggestion
minimum-version = "0.10"
```

**Review by bdice — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks all! 🙏

**Pull request merged**